### PR TITLE
Expand set of traced functions (ratelimit, backend/gce, GCEapi)

### PR DESCRIPTION
**Pull request by Lyoness**
Refs https://github.com/travis-ci/reliability/issues/199

## What is the problem that this PR is trying to fix?

See the steps and latency of things in our 
* backend/gce infra, including GCEAPI rate limiting
* redis rate limiting

## What approach did you choose and why?
Extension of  #516 

## How can you test this?
Running worker locally and viewing the traces on stackdriver staging project.  An example of an expanded trace is [here](https://console.cloud.google.com/traces/traces?project=travis-staging-1&organizationId=928883094116&q=%2Bapp:worker&tid=dbc795f7d2ec1089f71e12b013ec2c3c&start=1539812529379&end=1539816129379)

## What feedback would you like, if any?

what expanded traces would be useful, what would be noise? 
I tried to trace http requests in image.makeImageRequest in `api_selector.go` , but ran into some problems.

### CODE REVIEW

File: amqp_job.go [hunk 1]
```diff
@@ -63,6 +67,9 @@ func (j *amqpJob) Error(ctx gocontext.Context, errMessage string) error {
 }
 
 func (j *amqpJob) Requeue(ctx gocontext.Context) error {
+	ctx, span := trace.StartSpan(ctx, "amqpJob.Requeue")
+	defer span.End()
```
Comment (igorwwwwwwwwwwwwwwwwwwww) [inline amqp_job.go]: This is great, it might allow us to search for jobs that were requeued.

File: backend/gce.go [hunk 2]
```diff
@@ -578,6 +579,9 @@ func newGCEProvider(cfg *config.ProviderConfig) (Provider, error) {
 }
 
 func (p *gceProvider) apiRateLimit(ctx gocontext.Context) error {
+	ctx, span := trace.StartSpan(ctx, "apiRateLimit")
+	defer span.End()
```
Comment (igorwwwwwwwwwwwwwwwwwwww) [inline backend/gce.go]: This one should tell us how much time is being spent in the rate limiter! 🎉

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
This is awesome! I think we can drop a few of them but most of these are already showing stuff I wanted to know! :)

File: amqp_job.go [hunk 3]
```diff
@@ -91,6 +101,8 @@ func (j *amqpJob) Received(ctx gocontext.Context) error {
 }
 
 func (j *amqpJob) Started(ctx gocontext.Context) error {
+	ctx, span := trace.StartSpan(ctx, "amqpJob.Started")
+	defer span.End()
```
Comment (igorwwwwwwwwwwwwwwwwwwww) [inline amqp_job.go]: Maybe add a newline after this?

File: amqp_job.go [hunk 4]
```diff
@@ -99,6 +111,8 @@ func (j *amqpJob) Started(ctx gocontext.Context) error {
 }
 
 func (j *amqpJob) Finish(ctx gocontext.Context, state FinishState) error {
+	ctx, span := trace.StartSpan(ctx, "amqpJob.Finished")
+	defer span.End()
```
Comment (igorwwwwwwwwwwwwwwwwwwww) [inline amqp_job.go]: Maybe also a newline here?

File: backend/gce.go [hunk 5]
```diff
@@ -1337,6 +1372,8 @@ type warmerResponse struct {
 }
 
 func (i *gceInstance) sshConnection(ctx gocontext.Context) (remote.Remoter, error) {
+	ctx, span := trace.StartSpan(ctx, "GCE.sshConnection")
+	defer span.End()
```
Comment (igorwwwwwwwwwwwwwwwwwwww) [inline backend/gce.go]: Newline here as well.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Maybe also add a CHANGELOG entry under "Changed"?

**Comment by Lyoness** [comment]
Newest trace after changes is [here](https://console.cloud.google.com/traces/traces?project=travis-staging-1&organizationId=928883094116&q=%2Bapp:worker&tid=fcb12325066769a1479160bb62b35079&start=1539846738359&end=1539861138359)

**Pull request merged**